packages/ses/src/make-hardener.js

```diff
@@ -88,27 +107,25 @@ export const makeHardener = () => {
 
         // get stable/immutable outbound links before a Proxy has a chance to do
         // something sneaky.
-        const path = paths.get(obj) || 'unknown';
+        const path = weakmapGet(paths, obj) || 'unknown';
         const descs = getOwnPropertyDescriptors(obj);
         const proto = getPrototypeOf(obj);
         enqueue(proto, `${path}.__proto__`);
 
-        ownKeys(descs).forEach(name => {
+        arrayForEach(ownKeys(descs), (/** @type {string | symbol} */ name) => {
           const pathname = `${path}.${String(name)}`;
-          // todo uncurried form
-          // todo: getOwnPropertyDescriptors is guaranteed to return well-formed
+          // The 'name' may be a symbol, and TypeScript doesn't like us to
+          // index arbitrary symbols on objects, so we pretend they're just
+          // strings.
+          const desc = descs[/** @type {string} */ (name)];
+          // getOwnPropertyDescriptors is guaranteed to return well-formed
           // descriptors, but they still inherit from Object.prototype. If
           // someone has poisoned Object.prototype to add 'value' or 'get'
           // properties, then a simple 'if ("value" in desc)' or 'desc.value'
           // test could be confused. We use hasOwnProperty to be sure about
           // whether 'value' is present or not, which tells us for sure that
           // this is a data property.
-          // The 'name' may be a symbol, and TypeScript doesn't like us to
-          // index arbitrary symbols on objects, so we pretend they're just
-          // strings.
-          const desc = descs[/** @type {string} */ (name)];
-          if ('value' in desc) {
-            // todo uncurried form
+          if (objectHasOwnProperty(desc, 'value')) {
```

Comment:
A non-mechanical detail here is reärrangement of the comments to clarify their respective subjects. This was a point of confusion in @dominictarr’s review.